#! /bin/sh /usr/share/dpatch/dpatch-run
## 13-fix-fox-errors.dpatch by Thomas Girard <thomas.g.girard@free.fr>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: fix FOX compilation bugs

@DPATCH@
--- ACE_wrappers.orig/ace/FoxReactor/FoxReactor.cpp
+++ ACE_wrappers/ace/FoxReactor/FoxReactor.cpp
@@ -123,17 +123,17 @@
   ACE_Select_Reactor_Handle_Set dispatch_set;
   bool f=false;
   if(sel==SEL_IO_READ){
-    dispatch_set.rd_mask_.set_bit(ACE_HANDLE(handle));
+    dispatch_set.rd_mask_.set_bit(ACE_HANDLE((FXival)handle));
     f=true;
   }
   else
   if(sel==SEL_IO_WRITE){
-    dispatch_set.wr_mask_.set_bit(ACE_HANDLE(handle));
+    dispatch_set.wr_mask_.set_bit(ACE_HANDLE((FXival)handle));
     f=true;
   }
   else
   if(sel==SEL_IO_EXCEPT){
-    dispatch_set.ex_mask_.set_bit(ACE_HANDLE(handle));
+    dispatch_set.ex_mask_.set_bit(ACE_HANDLE((FXival)handle));
     f=true;
   };
   if(f) dispatch (1, dispatch_set);
@@ -248,9 +248,9 @@
 
   if (max_wait_time != 0)
     {
-      float t = max_wait_time->sec ()
+      float t = max_wait_time->sec () +
         max_wait_time->usec () / 1000000.0F;
-      fxapp->addTimeout (t*1000, this, 0);
+      fxapp->addTimeout (this, 0, t*1000);
     }
 }
 
--- ACE_wrappers.orig/TAO/tao/FoxResource/FoxResource_Loader.h
+++ ACE_wrappers/TAO/tao/FoxResource/FoxResource_Loader.h
@@ -22,6 +22,8 @@
 # pragma once
 #endif /* ACE_LACKS_PRAGMA_ONCE */
 
+#include <fx.h>
+
 #include "tao/Versioned_Namespace.h"
 
 TAO_BEGIN_VERSIONED_NAMESPACE_DECL
